### DISCUSSION TITLE
Import Measurements via USB from the Libre2 Standalone Reader

### DISCUSSION THREAD

**Discussion by yoshimo** [Navid200/xDrip#97] [Ideas]
The Freestyle Libre 2 optionally comes with a standalone device to scan your sensor patch.
You can use a usb cable to read out its data into a software running on your computer.
If you switch from the reader to the smartphone app or you don't have an nfc capable phone , which cheaper phones sometimes do, importing historic data into xdrip via usb would be great.

Maybe https://github.com/glucometers-tech/glucometerutils would help

**Comment by simonpickering** [comment]
The question is whether this needs to be a real-time interaction or something to add past data for analysis (which XDrip+ doesn't really do for "properly" historic data).

For example I have a fair few years worth of CSV data exported from various blood glucose meters, I also have Fitbit exports and custom food/drink logs exported as CSV. It could be useful to import historic data in a standarised form (i.e. I'll need to write something to get my particular flavours of data into the right format) assuming the data is useful for some historic data analysis purpose (might even be useful enough as a record keeping method of course).

OTOH if you're talking about near real-time import (plugging the meter in nearly each time you scan), then supporting the USB protocol becomes a priority, though I can't see that being a very fun way to live, especially when you can use your phone via Bluetooth/NFC. But perhaps there is a usecase there I don't see?

Nevertheless (who am I to decide!) I think it would be worth structuring to allow any/all/more of Bluetooth CGM, historic (CSV) data, data from USB connections, whatever else, to provide data to be logged and used, however they/these are imported.

**Comment by yoshimo** [comment]
I used to do measurements bloody with an accu chek, which connects fine with xdrip+ and backfills historic data.
Now that i have a freestyle libre2 , this isn't possible any more.
Without a NFC capable phone, i have an import issue at hand, real time or not , to save data.
This is kind of a fallback until we find a better way to use the cheap phone without nfc to connect the phone over bluetooth.

**Comment by Navid200** [comment]
We have discussions now opened in the main repository:
https://github.com/NightscoutFoundation/xDrip/discussions

I am looking for a way to transfer the open discussions from here to there. But, I'm not sure I'll succeed. In that case, it may be easier to open new discussions there.
